.github/workflows/integration.yaml

```diff
@@ -70,3 +70,100 @@ jobs:
           go run ./cmd/wazero run -hostlogging=filesystem -mount=.:/ ${{ steps.zig-build-test-binary.outputs.ZIG_BINARY_PATH }}
         # TODO: remove when #1029 is complete
         continue-on-error: true
+
+  tinygo:
+    name: TinyGo
+    runs-on: ubuntu-20.04
+    env:
+      # Until https://github.com/tinygo-org/tinygo/pull/3399 is released in 0.27.0,
+      # use the artifact from the GitHub Actions on the TinyGo repository.
+      # 3961571741 matches the one of the merge commit for the PR ^.
+      # https://github.com/tinygo-org/tinygo/actions/runs/3961571741
+      TINYGO_RUN_ID: 3961571741
+
+    steps:
+      - uses: actions/cache@v3
+        id: cache
+        with:
+          path:
+            ~/.cache/go-build
+            ~/.cache/tinygo
+          key: integration-test-tinygo-${{ env.TINYGO_RUN_ID }}-${{ env.GO_VERSION }}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Install TinyGo
+        env:
+          # Necessary to use gh command.
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh run download --repo tinygo-org/tinygo ${{ env.TINYGO_RUN_ID }} -n linux-amd64-double-zipped
+          tar -xzf tinygo.linux-amd64.tar.gz
+          sudo dpkg -i tinygo_amd64.deb
+
+      - name: Checkout wazero
+        uses: actions/checkout@v3
+
+      - name: Install wazero
+        run: go install ./cmd/wazero
+
+      - name: Generate target json for wazero.
+        run: |
+          jq '.emulator = "wazero run -hostlogging=filesystem -mount=.:/ -mount={tmpDir}:/tmp {}"' \
+            $(tinygo env TINYGOROOT)/targets/wasi.json > ./wasi_wazero.json
+
+      - name: Run standard library tests
+        # On PRs, we only test the minimal std packages to test basic functionality.
+        run: tinygo test -target ./wasi_wazero.json os math hash
```

Comment:
any specific reason for math and hash? `os` is a great one as it makes a lot of sys calls.

One of the following would also be good, if affordable, as they use relative dir lookups. If needed we could swap out math or hash for one?
```
./src/compress/flate:/ ../testdata ../../testdata
./src/image:../testdata
./src/net:/ /etc:/etc
./src/text/template/:/ /tmp:/tmp
```

Reply:
`image` and `text/template` are not even supported by TinyGo itself, and `ompress/flate` is not test target package for Wasm target in TinyGo. So, I swapped hash with the only left `net` .